app/src/main/java/org/mozilla/fenix/settings/address/ext/Address.kt

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.address.ext
+
+import androidx.annotation.VisibleForTesting
+import mozilla.components.concept.storage.Address
+
+/**
+ * Generate a label from a given [Address]. A well-formed [Address] will have
+ * a full name in the label, otherwise it will find the highest priority [Address] property
+ * to use, as based on the desktop code found in FormAutofillUtils::getAddressLabel
+ * https://searchfox.org/mozilla-central/rev/87ecd21d3ca517f8d90e49b32bf042a754ed8f18/toolkit/components/formautofill/FormAutofillUtils.jsm#323
+ */
+fun Address.toAddressLabel(): String = getFullName().ifEmpty { iterateFields("") }
```

Comment:
I feel we might be overthinking this with trying to generate a well formed label and description depending on what fields are available. The UI here is very similar to Chrome's Addresses where it only shows the full name in the label and all other address field appears as the description.

For the time being to get this all landed, I think we should simply call `getFullName` for the label and drop `toAddressLabel`. 

If we did decide to go with your methodology, I think I would like to see something very explicitly stated in a issue and okay'd by UX/Product.

Reply:
🤦 I'd been basing my work on what I saw on desktop instead of the current designs. I'll update all this accordingly today. Thanks for your patience in pointing that out

Reply:
Okay, the one reasoning why I was kinda strict about looking at seeing what Desktop was doing was to understand if and how they did the address ordering for different regions. It was better that we understood what was done on Desktop and be informed as we did our own mobile implementation. 

Any implicit requirements that we might get from Desktop should be brought up with UX/Product if we felt it needed further consideration.